app/scripts/background.js

```diff
@@ -463,7 +455,9 @@ function setupController(initState, initLangCode, remoteSourcePort) {
    *
    * @param {Port} remotePort - The port provided by a new context.
    */
-  function connectRemote(remotePort) {
+  async function connectRemote(remotePort) {
+    await isInitialized;
```

Comment:
This should guarantee that `connectRemote` always works correctly when called. It now waits until initialization has completed.

But this PR is still broken because the `connectRemote event listener is still added too late in the process. We don't listen until after initialization.

Reply:
Fixed by https://github.com/MetaMask/metamask-extension/pull/16559/commits/7038e6adbb70a6a50b416f5d11e827268f4c280e